nifi-nar-bundles/nifi-hashicorp-vault-bundle/nifi-hashicorp-vault-client-service-api/src/main/java/org/apache/nifi/vault/hashicorp/HashiCorpVaultClientService.java

```diff
@@ -0,0 +1,152 @@
+                    "Environment Configuration documentation (https://docs.spring.io/spring-vault/docs/2.3.x/reference/html/#vault.core.environment-vault-configuration). " +
+                    "All of the Spring property keys and authentication-specific property keys are supported.")
+            .required(true)
+            .addValidator(MultiFileExistsValidator.INSTANCE)
+            .dependsOn(CONFIGURATION_STRATEGY, PROPERTIES_FILES)
+            .identifiesExternalResource(ResourceCardinality.MULTIPLE, ResourceType.FILE)
+            .build();
+
+    PropertyDescriptor CONNECTION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("vault.connection.timeout")
+            .displayName("Connection Timeout")
+            .description("The connection timeout for the HashiCorp Vault client")
+            .required(false)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    PropertyDescriptor READ_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("vault.read.timeout")
+            .displayName("Read Timeout")
+            .description("The read timeout for the HashiCorp Vault client")
+            .required(false)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    /**
+     *
+     * @return A service for communicating with HashiCorp Vault.
+     */
+    HashiCorpVaultCommunicationService getHashiCorpVaultCommunicationService();
+
+    class MultiFileExistsValidator implements Validator {
+
+        private static final Validator INSTANCE = new MultiFileExistsValidator();
+
+        @Override
+        public ValidationResult validate(final String subject, final String value, final ValidationContext context) {
+            String reason = null;
+            if (value == null) {
+                reason = "At least one file must be specified";
+            } else {
+                final Set<String> files = Arrays.stream(value.split(","))
+                        .map(String::trim)
+                        .collect(Collectors.toSet());
+                try {
+                    for (final String filename : files) {
+                        final File file = new File(filename);
+                        if (!file.exists()) {
+                            reason = "File " + file.getName() + " does not exist";
+                        } else if (!file.isFile()) {
+                            reason = "Path " + file.getName() + " does not point to a file";
+                        }
+                    }
+                } catch (final Exception e) {
+                    reason = "Value is not a valid filename";
+                }
+            }
+
+            return new ValidationResult.Builder().subject(subject).input(value).explanation(reason).valid(reason == null).build();
+        }
+    }
```

Comment:
This custom validator should not be necessary with the `identifiesExternalResource()` feature on the Property Descriptor.